File: frontend/src/types/source.ts
export const SOURCE_TYPES = [
  "job_board",
  "vc_portfolio",
  "government",
  "news",
  "directory",
  "regulatory",
  "environment",
  "research",
  "custom",
] as const;

export type SourceType = (typeof SOURCE_TYPES)[number];

export const SOURCE_STRATEGIES = [
  "yc",
  "generic_jobs",
  "generic_portfolio",
  "playwright_portfolio",
  "hn_hiring",
  "generic_table",
  "generic_list",
  "rss_feed",
  "pdf_extract",
  "playwright_spa",
  "api_json",
] as const;

export type SourceStrategy = (typeof SOURCE_STRATEGIES)[number];

export type SourceHealth = "ok" | "warning" | "dead";

export interface Source {
  id: string;
  name: string;
  url: string;
  type: SourceType;
  strategy: SourceStrategy;
  enabled: boolean;
  lastScraped: string | null;
  itemCount: number;
  health: SourceHealth;
  notes: string | null;
  lastSuccessAt: string | null;
  lastFailureAt: string | null;
  failureReason: string | null;
}

export interface SourceCreate {
  name: string;
  url: string;
  type: SourceType;
  strategy: SourceStrategy;
}

File: backend/schemas/source.py
"""Source request/response schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict

from schemas.enums import SourceHealth, SourceStrategy, SourceType


class SourceCreate(BaseModel):
    name: str
    url: str
    type: SourceType = SourceType.CUSTOM
    strategy: SourceStrategy = SourceStrategy.GENERIC_LIST
    notes: str | None = None


class SourceOut(BaseModel):
    id: str
    name: str
    url: str
    type: str
    strategy: str
    enabled: bool
    last_scraped: datetime | None
    job_count: int
    health: str
    notes: str | None

    model_config = ConfigDict(from_attributes=True)
